src/hooks/useChromeLocalStorage.js

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react'
+import { writeToLocalStorage, readFromLocalStorage } from '../api/chrome'
+
+export function useChromeLocalStorage(key, initialValue) {
+  const [data, setData] = useState(initialValue)
+  const [localData, setLocalData] = useState(data)
+  const [hasError, setHasError] = useState(false)
+
+  useEffect(() => {
+    if (data) {
+      setLocalData(data)
+    }
+  }, [data])
+
+  useEffect(() => {
+    const readLocalStorage = async () => {
+      try {
+        const value = await readFromLocalStorage(key)
```

Comment:
I think it's easier if you use the `then/catch` instead of the async. Isn't it?

Reply:
no let's keep async please, we prefer to do it this way